flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/ReadableCatalog.java

```diff
@@ -224,4 +224,54 @@ CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partiti
 	 */
 	boolean functionExists(ObjectPath functionPath) throws CatalogException;
 
+	// ------ Statistics ------
+
+	/**
+	 * Get the statistics of a table.
+	 *
+	 * @param tablePath path of the table
+	 * @return the statistics of the given table
+	 *
+	 * @throws TableNotExistException if the table does not exist in the catalog
+	 * @throws CatalogException	in case of any runtime exception
+	 */
+	CatalogTableStatistics getTableStatistics(ObjectPath tablePath) throws TableNotExistException, CatalogException;
```

Comment:
What happens if this API is called on a partitioned table?

Reply:
Partitioned table has its own, table-level stats, though it may not be complete or accurate, as in Hive's case.